src/input/sources/hls.rs

```diff
@@ -0,0 +1,139 @@
+use std::{
+    io::{ErrorKind as IoErrorKind, Result as IoResult, SeekFrom},
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use bytes::Bytes;
+use futures::StreamExt;
+use pin_project::pin_project;
+use reqwest::{header::HeaderMap, Client};
+use serenity::async_trait;
```

Comment:
```suggestion
use async_trait::async_trait;
```

Songbird can be built as a pure audio server, so serenity isn't always a valid dependency.